docs/configuration/index.md

```diff
@@ -1481,9 +1481,35 @@ These Broker configurations can be defined in the `broker/runtime.properties` fi
 |`druid.broker.select.tier`|`highestPriority`, `lowestPriority`, `custom`|If segments are cross-replicated across tiers in a cluster, you can tell the broker to prefer to select segments in a tier with a certain priority.|`highestPriority`|
 |`druid.broker.select.tier.custom.priorities`|`An array of integer priorities.`|Select servers in tiers with a custom priority list.|None|
 
+##### Query laning
+
+*Laning strategies* allow you to control capacity utilization for heterogeneous query workloads. With laning, the broker examines and classifies a query for the purpose of assigning it to a 'lane'. Lanes have capacity limits, enforced by the broker, that can be used to ensure sufficient resources are available for other lanes or for interactive queries (with no lane), or to limit overall throughput for queries within the lane. Requests in excess of the capacity are discarded with an HTTP 429 status code.
+
+|Property|Description|Default|
+|--------|-----------|-------|
+|`druid.query.scheduler.numThreads`|Maximum number of HTTP threads to dedicate to query processing. To save HTTP thread capacity, this should be lower than `druid.server.http.numThreads`.|Unbounded|
```

Comment:
in what use case would I ever want to set it something other  than `druid.server.http.numThreads` ? If the recommendation is to set it lower than `druid.server.http.numThreads` then why the default value is not set to `druid.server.http.numThreads - 1` ?
I guess, as a user I don't quite understand the importance of setting this higher/same/lower compared to `druid.server.http.numThreads` and when I should choose one vs the other.

Reply:
>in what use case would I ever want to set it something other than druid.server.http.numThreads ?

I actually think we might always want to set it lower than `druid.server.http.numThreads`, but I was too nervous to make this the default and made it opt in behavior instead (since it grabs and releases locks for each query if there is some bug in releasing locks a broker would eventually stop accepting queries entirely). The primary reason I think we want it lower than `druid.server.http.numThreads` is to save some 'slack' space for non-query http connections, like accepting health checks, lookup management, and other such things that can be starved when long running queries start to pile up.

>If the recommendation is to set it lower than `druid.server.http.numThreads` then why the default value is not set to `druid.server.http.numThreads - 1` ?

See my above nervousness, but I think `druid.server.http.numThreads - 1` would probably be a good default. This might want to be adjusted to be even lower depending on how much other non query http traffic the server is receiving (e.g. frequently polled/updated lookups, etc).

>I guess, as a user I don't quite understand the importance of setting this higher/same/lower compared to druid.server.http.numThreads and when I should choose one vs the other.

I would agree the current documentation doesn't quite adequately describe how this stuff might be utilized, in a future PR i want to add a section to [cluster tuning docs](https://github.com/apache/druid/blob/master/docs/operations/basic-cluster-tuning.md) to more properly advise on when and how to set this stuff up.

Reply:
thanks, I understand the  reasoning now.
lookup end points already have a QoS filter to never consume more than two threads from jetty, I wonder if in this world it makes sense to setup QoS filter for non-query endpoints(say hardcoded to 2) so that we can ensure that  they don't end up consuming more jetty threads than intended.
then default  `druid.query.scheduler.numThreads = druid.server.http.numThreads - numReservedForOthers=4` and users would likely never be expected to touch `druid.query.scheduler.numThreads` .

Major behavior change with lane usage is really losing the queuing  of requests to handle spikes and instead sending 429s immediately.  In future, we  could introduce mechanism to maintain statically/dynamically sized [per lane] waiting  queue ourselves as well along with concurrency limits in lane strategy.

Reply:
I wonder if it would make sense to instead move towards automatically computing `druid.server.http.numThreads`, since maybe it is easier for operators to only have to think about the number of concurrent queries to serve and just set `druid.query.scheduler.numThreads`?

>Major behavior change with lane usage is really losing the queuing of requests to handle spikes and instead sending 429s immediately. In future, we could introduce mechanism to maintain statically/dynamically sized [per lane] waiting queue ourselves as well along with concurrency limits in lane strategy.

Yeah the current behavior is definitely a hard stop if you are over the line. I agree it would make sense to allow some sort of timed out queuing behavior, which is what jetty QoS filter can sort of provide, which is a large part of [why I am still wondering](https://github.com/apache/druid/pull/9407#discussion_r389159233) if `druid.query.scheduler.numThreads` should be a QoS filter instead of enforced as an actual lane like it is currently.

Reply:
It is fine to let user provide `druid.query.scheduler.numThreads` and compute `druid.server.http.numThreads` , just that one of those should not be touched by user in most cases.

There are few advantages in maintaining the queues ourselves and not letting jetty do it.
- we have no control over jetty queue, if a request is dropped then end  user sees that as a TCP connection close and not a HTTP 429. So, to client, it is not clear whether to retry or backoff.
- we don't know how much time request waited  in jetty queue, consequently request time metrics don't account for that.
- jetty queue is [probably] static in size, if we managed it ourselves then we have the option of keeping dynamically sized queues and do potentially other cool things.